### PR TITLE
Add agent string to IConnect/IConnected messages and log in telemetry.

### LINKED ISSUE #7598: Communicate agent string to SPO/PUSH
It is very useful for PUSH to know a version of a client that connected to it, such that it can record it in its own telemetry.
This would allow us to access qualify of client + server bits. I.e. once bits are in DF (pre-production ring), we can see if number of warnings / errors recorded by PUSH is higher for this client version compared to prior version, and thus make decision on further rollout.

Same is true in reverse - it would be great for client to receive and record PUSH version in client telemetry (for same reason - see if some client metrics got better or worse due to a chance on service side).

Each side can make independent decision what "version" is (i.e. what kind of information is there / format), for example client may record versions of driver + loader. But to the other party it's just an agent string.

We should add it to IConnected / IConnect as unique fields.

I'm not sure about SPO calls. Client today does not report (in client telemetry) info about every SPO call, so even if SPO was exposing this info, we would not have place to record it. I'd say that SPO workflow is lower priority and out of scope for now, i.e. focus of this task is for PUSH workflows.

Same is true for R11S.

### PR DESCRIPTION
**Pull request by jatgarg**
Fixes: https://github.com/microsoft/FluidFramework/issues/7598
Add agent string to IConnect message for push to log in telemetry and log relayServiceAgent in connectionStateProps so that they get logged in telemetry on error.

### CODE REVIEW

File: packages/drivers/odsp-driver/src/odspDocumentDeltaConnection.ts [hunk 1]
```diff
@@ -226,6 +227,8 @@ export class OdspDocumentDeltaConnection extends DocumentDeltaConnection {
             versions: protocolVersions,
             nonce: uuid(),
             epoch: epochTracker.fluidEpoch,
+            relayUserAgent: client.details.environment !== undefined ?
+                `${client.details.environment};driverVersion:${pkgVersion}` : `driverVersion:${pkgVersion}`,
```
Comment (vladsud) [inline packages/drivers/odsp-driver/src/odspDocumentDeltaConnection.ts:231]: space after ; would be nice - a bit more readable (by humans, though requires trimming by bots :))

File: packages/loader/container-loader/src/deltaManager.ts [hunk 2]
```diff
@@ -1236,6 +1236,9 @@ export class DeltaManager
             clientId: connection.clientId,
             mode: connection.mode,
         };
+        if (connection.relayServiceAgent !== undefined) {
+            this.connectionStateProps.relayServiceAgent = connection.relayServiceAgent;
```
Comment (vladsud) [inline packages/loader/container-loader/src/deltaManager.ts:1240]: Is there any way to get this property to _Connected event in container? I believe this.connectionStateProps we use in very few places, mostly to debug connectivity issues. Take a look at DeltaManager.connectionProps

File: packages/drivers/odsp-driver/src/odspDocumentDeltaConnection.ts (hunk 1 above)
Comment (vladsud) [inline packages/drivers/odsp-driver/src/odspDocumentDeltaConnection.ts:231]: I'm not sure if it works everywhere same as it works in Node, but [client.details.environment, `driverVersion:${pkgVersion}`].join("; ") seems to give what we want even if using undefined, and it feels shorter way to express what we want to do (though less efficient at execution :)).

**Review by vladsud — APPROVED** [review]

File: packages/loader/container-loader/src/deltaManager.ts (hunk 2 above)
Comment (jatgarg) [reply packages/loader/container-loader/src/deltaManager.ts:1241]: Added there.

**Pull request merged**